### ISSUE TITLE
New cite-key structure

### ISSUE THREAD

**Issue by nirbhasa** [dharmiks/sri-chinmoy-library-cs#2]
Hi @dharmik, We have slightly modified the cite key for top level books. Also new URLs now reflect the cite-key, so they stay the same for the next few decades. 

Priyadarshan's excellent explanation : 

A cite-key could be seen in two separate ways, the usual, science-related academia way, and the more specific humanistic classic-related scholarly way.

The latter is the one suggested by late Prof. Lambert, the real planner for the whole structure. He felt Sri Chinmoy's works should really be seen as classics, now and forever, and thus, they should belong to the second (more "elite") category of bibliographic schema of cite-keys.

In that case, there is no such thing a "book-in-a-series".

We do still have cite-keys for those series, but just for our own archiving needs. In reality, all books pertaining to a series should have one cite-key only.

For example, Sri Chinmoy answers, book 1 to 38, would all have cite-key: SCA, since the works is "Sri Chinmoy answers", not how the initial publisher happened to print the individual volumes.

But it is all right we still keep a volume in the cite-key. It was allowed, although, in proper scholarly books, it should not be referred like that, but rather as "SCA" for the work, and, for example, "SCA 44" for the unit.

Thus, if we want to remain faithful to the original idea and design of the bibliography (and the archives' architecture), the best way to convey this to the final reader is to give proper importance to the Unit cite key first and foremost, and only second to the books.

Since "dash" is more common to type and to spell that "underscore", and units are generally more linked to, in respect to whole books, I would perhaps consider using this system:

Cite key for unit: "SCA 727", linked as "/SCA-727" (when part of URL)
Cite-key for volume-in-a-series: "SCA volume 1", linked as "/SCA_1" (when part of URL)

The latter is really for our own internal convenience (for the site), but it should not prevaricate the former, which is the real, main backbone of the whole structure.

---

So, in light of this, we need to decide what to do with foreign language cite keys and permalinks, they are currently quite confused. Here is my suggestion:

Top level books: xxx-cs volume 1 (or whatever the Czech for volume is) permalink xxx-cs_1
Book pages: xxx-cs 1 , permalink xx-cs-1

It would be good to have feedback also from Priyadarshan

**Comment by dharmiks** [comment]
I agree with this idea and I think the only right way is to keep the same structure for all languages. Just tell me how do you plan to change it - by script or manually and when? Recently I am working on czech archive.
Also we should set properly postfixes like common sense in czech and slovak:
cz = czech
sk = slovak
cs = czechoslovak

**Comment by nirbhasa** [comment]
Hi, am about to bring on line these changes today. One thing I have kept for the moment is the existing language prefixes 'cs' (czech) 'sk' (Slovak), as they correspond to the official international language codes for those languages (so in order to set page language, I would have to use these codes anyway. So it is probably better to keep them in the org-mode files to avoid confusion)

There is no language code for Czechoslovak. For 'czechoslovak' books - are these for books that will appear on both Czech and Slovakian sites? Do you have any of these at the moment?

**Comment by nirbhasa** [comment]
Ps - for now you don't need to do anything with cite keys in the org mode file. This is my current plan
- Make sure books with old cite key show the new cite key and URL when they are imported (done)
- Work on the cite key script so that it works with new format
- There might be some files you need to do manually (e.g. aspiration plants) but you can do at your leisure

**Comment by priyadarshan** [comment]
Hello, I was told about this ticket by Nirbhasa, as I am not automatically notified of all tickets.

We did discuss it last year (was it at Udine's Joy Day?), but since then, it came to me that we could leverage this need for additional functionality.

If I may submit another possibility, it might be useful to prepend the language code, instead of appending it.

Something like:

`cs_SCA-27`
or
`es_EA-13`

The semantic function would still be there, and all cite-keys would be sorted by language, if one would ever need it.

**Comment by nirbhasa** [comment]
> On 18 Mar 2015, at 12:41, Priyadarshan notifications@github.com wrote:
> 
> Hello, I was told about this ticket by Nirbhasa, as I am not automatically notified of all tickets.
> 
> We did discuss it last year (was it at Udine's Joy Day?), but since then, it came to me that we could leverage this need for additional functionality.
> 
> If I may submit another possibility, it might be useful to prepend the language code, instead of appending it.
> 
> Something like:
> 
> cs_SCA-27
> or
> es_EA-13
> 
> The semantic function would still be there, and all cite-keys would be sorted by language, if one would ever need it.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/dharmiks/sri-chinmoy-library-cs/issues/2#issuecomment-82952675.

Thank you Priyadarshan. I have also filed quite a few queries regarding cite keys under the archives project, I am not sure if you are automatically notified of those as well

**Comment by priyadarshan** [comment]
If it is not indispensable for the website, I would even consider _not_ adding the language prefix (or suffix), since that can be done at any time, by taking the language code, at the top of each book file, and adding a given cite-key.

A website application could build a proper "navigational" cite-key on the fly, without forcing the editors to  use a very long or awkward structure.

This soultion (ie, no language code at all in cite-keys), is the one I personally consider stronger from the Data architecture point of view.

**Comment by nirbhasa** [comment]
> On 18 Mar 2015, at 13:33, Priyadarshan notifications@github.com wrote:
> 
> If it is not indispensable for the website, I would even consider not adding the language prefix (or suffix), since that can be done at any time, by taking the language code, at the top of each book file, and adding a given cite-key.
> 
> A website application could build a proper "navigational" cite-key on the fly, without forcing the editors to use a very long or awkward structure.
> 
> This soultion (ie, no language code at all in cite-keys), is the one I personally consider stronger from the Data architecture point of view.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/dharmiks/sri-chinmoy-library-cs/issues/2#issuecomment-82972716.

It might even suit the website better to have it this way, from a URL structure point of view

English book: http://www.srichinmoylibrary.com/tdy
Czech book: http://cz.srichinmoylibrary.com/tdy (Czech version)

@priyadarshan: When displaying the cite key on the site (or any other deployment), would you display it as TDY, or add the language suffix and display as TDY-cs? Does it matter?

**Comment by priyadarshan** [comment]
Yes, those examples display the way I said above, and to me feel the better choice.

As for cite-keys, as long as we refer to the "official" translation (ie the one approved by the Centre as representative of the original text), there would be used as tdy (or TDY), not mentioning the language explicitly, as it would be clear from the context.

**Comment by dharmiks** [comment]
A few comments to Czechoslovak pages:
- for public on web should exist just one czechoslovak page : http://cs.srichinmoylibrary.com, 
- manual selection of language czech and slovak,
- automatic language for page changed according place of user (from Czech rep. czech, from Slovak slovak, from the rest of the world czech)
- just now are all books and covers in one common czechoslovak folders cz-sri-chinmoy-archives, cz-archives-covers (can be splitted or renamed to cs-), files have postfixes -cs or -sk

**Comment by nirbhasa** [comment]
Hi Dharmik, spoke with Tejvan about this
- in the immediate future, what we can do is also list Czech books on Slovak pages and vice-versa. This will solve problem of Slovak site looking quite bare :)
- you talked about maybe having own Czechoslovakian site. We are happy with this, my only concern is if the effort is not better spent making a better site for everyone. Myself and Tejvan discussed some ideas on involving other people to make this happen. I can explain more over weekend.

**Comment by nirbhasa** [comment]
Hi Dharmik, after all the discussion about cite keys, it seems the cite keys in the original English org mode files will not change at all. How they are displayed on the web site, and in the URL will change, but not the files themselves.

So for Czech, it is best to just leave everything as it is for now in the files themselves (including -cs). On the website, the only thing that will change is that we will take the ‘-cs’ out of the permalink so that it will be the same as the English original

eg Today
http://www.srichinmoylibrary.com/tdy (English)
http://sk.srichinmoylibrary.com/tdy (Slovak)

Also, I think for foreign languages I will leave out the bit below the page about ’This page can be referenced using cite-key… ‘. Do you have any real need for it in Czech? I think academic referencing is mainly important for the English originals. 

This means that the cite key for foreign languages is just an ‘internal’ tool so we can distinguish pages. It also means that we don’t need to decide what the format will be for all time (unlike English). So lets just keep using the same format as we have been using for many months.

I changed 27000 Aspiration Plants, part 1 back to the original format

**Comment by nirbhasa** [comment]
Regarding the Czech and Slovak websites: I have put in place a simpler solution, and one which in my humble opinion is better.
- make Czech books available on the front page of Slovak site (can also add Slovak books to Czech site too). 
- For websites with not many books (e.g. Slovak) list books directly instead of alphabet listing
- I have also given people the option of seeing English originals on non English sites. 
- There are some language strings that need to be translated.
- We can definitely still add some browser language redirection.

**Comment by dharmiks** [comment]
Hi Nirbhasa,
- about ’This page can be referenced using cite-key… ‘ - on page could be somewhere info about cite-key, I am using it quite often now and maybe others also... .(I know, that this info I can get also from the main link, but new users do not know what cite-key is...)
